### PR TITLE
Add basic settings API for admins

### PR DESCRIPTION
**Pull request by SaptakS**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
Basic Implementation of Settings API

#### Changes proposed in this pull request:

- Since there will be only one settings entry, so as discussed with @niranjan94 we will make a blank entry on setup and then we will only be using ResourceDetail to PATCH, GET the API.
- Based on the admin access of the user, different Schemas are shown.

### CODE REVIEW

File: app/api/settings.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    analytics_key = fields.Str()
+
+
+class SettingDetail(ResourceDetail):
+    """
+    setting detail by id
+    """
+
+    def before_get(self, args, kwargs):
+        global show_admin_schema
```
Comment (shubham-padia) [inline app/api/settings.py]: @SaptakS What is this used for, I was unable to find this elsewhere in the diff

Comment (SaptakS) [inline app/api/settings.py]: Oops. Forgot to remove that. Thanks for pointing out

Comment (SaptakS) [inline app/api/settings.py]: Resolved.

Comment (shubham-padia) [inline app/api/settings.py]: :white_check_mark:

**Pull request merged**